### PR TITLE
feat: polygon migration interaction

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

- Uses ethers to load the migration details and status for the profile
- Updates the migrations section with the migrations that come from the hook
- updates the notifications dropdown to show the "pending migrations"

Notice that this PRs still hardcodes some  polygon transaction ids to ensure it shows migrations, also notice that you need to define new `env` variables when testing locally. For that use this info:

```
VITE_POLYGON_CONTRACT_ADDRESS=0x4a12a2ADc21F896E6F8e564a106A4cab8746a92f
VITE_POLYGON_RPC_URL=https://rpc-mumbai.maticvigil.com/
```

closes https://app.clickup.com/t/39xfbh6 and since I updated the notifications section also https://app.clickup.com/t/86773zpn8

## Checklist

-   [ ] My changes look good in both light AND dark mode
-   [ ] The change is not hardcoded to a single network, but has multi-asset in mind
-   [ ] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: src/domains/migration/pages/Migration/Migration.tsx [hunk 1]
```diff
@@ -53,11 +35,16 @@ export const Migration = () => {
...
 
 				<Section className="mt-4">
-					<MigrationTransactionsTable
-						migrationTransactions={migrations}
-						isCompact={isCompact}
-						onClick={() => console.log("row click")}
-					/>
+					{migrations === undefined ? (
+						// @TODO: add skeleton
+						<></>
```
Comment (alfonsobries) [inline src/domains/migration/pages/Migration/Migration.tsx:40]: Added a task for this https://app.clickup.com/t/865bgdg9m

File: src/app/contexts/Migration/Migration.tsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+// @TODO: make this transactions dynamic, an option is to store/get using the
+// env storage, e.g. env.storage().get("ark-migration-transactions")
+const fakeTransactionsIds = [
+	// Invalid one
+	"0x74a1e612846da4c3da618a77e82e6f8468fd88459662738babce5b6014b023f4",
+	// Valid ones
+	"0x2c00cb87957dd47ee9e6a03210bf2ca46ba7a11156391d3b5960760846de226a",
+	"0x5e4f8256caffe7d7276da0611bcf8a8bff8aea1219853ea07d9575355efff704",
+];
```
Comment (alfonsobries) [inline src/app/contexts/Migration/Migration.tsx:62]: ticket https://app.clickup.com/t/865bgdrfu

File: src/app/contexts/Migration/Migration.tsx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+		// Not sure yet how that will work but maybe we can use the profile to get the address
+		// and then the transactions or maybe we can save those values in the env storage
+		// when adding a new migration
+		address: "PeNdInGPeNdInGPeNdInGPeNdInG",
+		amount,
+		id: migration.arkTxHash,
+		migrationAddress: migration.recipient,
+		status,
+		// @TODO: read above
+		timestamp: Date.now() / 1000,
```
Comment (alfonsobries) [inline src/app/contexts/Migration/Migration.tsx:83]: Same ticket https://app.clickup.com/t/865bgdrfu

File: src/app/contexts/Migration/Migration.tsx [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+			// @TODO: Potentially remove this in favour of storing the transactions
+			// when a new migration is added
+			...fakeTransactionsIds,
+		]
+			// Remove duplicated transactions ids
+			// @TODO: not needed once we remove the fake ones
+			.filter((id, index, original) => original.indexOf(id) === index);
+		// @TODO: potential code to replace the one above once the fake
+		// transactions are removed
+		// const transactionsIds = migrations.map((tx: Migration) => tx.id);
```
Comment (alfonsobries) [inline src/app/contexts/Migration/Migration.tsx:142]: same ticket https://app.clickup.com/t/865bgdrfu

File: .env.example [hunk 5]
```diff
@@ -1,5 +1,7 @@
 VITE_POLYGON_MIGRATION_TRANSACTION_FEE=0.05
+VITE_POLYGON_CONTRACT_ADDRESS=0xContractAddress
+VITE_POLYGON_RPC_URL=https://rpc-mumbai.maticvigil.com/
+
```
Comment (ItsANameToo) [inline .env.example:4]: we'll need a variable for defining the polygonscan url as well, but will make that into another task so we can get this merged early on

File: src/app/contexts/Migration/Migration.tsx [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+			: MigrationTransactionStatus.Confirmed;
+
+	return {
+		// @TODO: get the address and timestamp
+		// Not sure yet how that will work but maybe we can use the profile to get the address
+		// and then the transactions or maybe we can save those values in the env storage
+		// when adding a new migration
+		address: "PeNdInGPeNdInGPeNdInGPeNdInG",
+		amount,
+		id: migration.arkTxHash,
```
Comment (ItsANameToo) [inline src/app/contexts/Migration/Migration.tsx:79]: this should link to the polygon ID instead. Currently the UI links to polygonscan with the ARK transaction ID instead

**Review by ItsANameToo — APPROVED** [review]

**Comment by ItsANameToo** [comment]
I'm creating cards for the comments @alfonsobries , that way we can get this PR merged

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**